src/Drupal/DrupalExtension/Context/DrupalContext.php

```diff
@@ -541,4 +553,73 @@ public function assertTextInTableRowInRegion($text, $rowText, $region): void {
     }
   }
 
+  /**
+   * Take a screenshot.
+   *
+   * @Then I take screenshot
+   */
+  public function iTakeScreenshot(): void {
+    $this->takeScreenshot(FALSE);
+  }
+
+  /**
+   * Take a screenshot on failure.
+   *
+   * @AfterStep
+   */
+  public function takeScreenshotOnFailure(AfterStepScope $scope): void {
+    // We cannot use !isPassed() as the scenario outline returns a non-PASSED code.
+    if ($scope->getTestResult()->getResultCode() === TestResult::FAILED) {
+      $this->takeScreenshot(TRUE);
+    }
+  }
+
+  /**
+   * Take screenshot or content html
+   *
+   * @var bool $on_failure
+   *   TRUE if called on failure.
+   */
+  public function takeScreenshot(bool $on_failure = FALSE): void {
+
+    $filepath = $this->getScreenshotParameter('path');
+    if (empty($filepath) || (!is_dir($filepath) && !mkdir($filepath, 0777, true) && !is_dir($filepath))) {
+       $filepath = sys_get_temp_dir();
+    }
+
+    $feature_file = $this->getFeature()->getFile();
+    list($feature_filename, $extension) = explode('.', substr($feature_file, strrpos($feature_file, '/') + 1));
+    $filename = sprintf("%s_%s_%s%s", date('mdy-His'), $feature_filename, $this->getStep()->getLine(), $on_failure ? $this->getScreenshotParameter('failure_suffix') : '');
+
+    $output_filepath = $filepath . "/" . $filename;
```

Comment:
@kumikoono This should be:

```suggestion
    $filename = sprintf('%s_%s_%s%s', date('mdy-His'), $feature_filename, $this->getStep()->getLine(), $on_failure ? $this->getScreenshotParameter('failure_suffix') : '');

    $output_filepath = "${filepath}/${filename}";
```

Reply:
Changed.